### PR TITLE
Ar/feature/dom updates

### LINKED ISSUE #8: DOM updates - modular architecture
As a developer,

When I am building out the structure (or refactoring architecture) for an app,

I want to separate the methods responsible for DOM-updates,

So that the project has a more modular file structure and is consequently more DRY&SRP.

- [x] create `domUpdates.js` in src directory (we don't need to test these methods here)
- [x] will need to import / export 

file structure will be similar to:

```
// domUpdates.js

let domUpdates = {
  updateDomMethod1(){...},
  updateDomMethod2(){...},
  ...
};

export default  domUpdates;
```

examples to import from separate file:
showMyRecipesBanner() on line 241
showWelcomeBanner() on line 246

### PR DESCRIPTION
**Pull request by srslie**
**Changes:**
- added helper functions 
    - event listeners
    - finding targetRecipe
- pulled out dom manipulation in domUpdates
    - toggling
    - adding html 
    - clearing html
    - displaying cards
    - displaying pantry
- made the enter, search button, tag searches work
- added to pantry
    - display of ingredients and amounts user has
    - form to add more/post to api
    - button to search which recipes users can make using their pantry ingredients
- added to card
     -  button to compare recipe against the users' ingredients

**Issues:**
- data is whack, giving us some errors for if ids/names/all sorts of properties are null or undefined
- doing some strange concatination because of a string/number type error that we also have with recipeID keys, so many of the logics use rough equality to check conditions
- We need to have some option to remove ingredients if the user want to manually from their pantry. I can make the min on the addingredient form negative if we take care of it that way, otherwise we have to design something.
- If a user adds a new ingredient that's not one we already know, we have no way to  retrieve the name because ingredients only have amounts and ids. Also, we only ever map common names in the user pantry ingredients, so we can't match those names easily against the ingredients when we go to update the server.
- The button to compare recipes isn't working, mainly because I'm having a hard time bubbling the event and maybe it needs to not be a button or I need to add event listeners based on a querySelectorAll after each time cards are created
- Do we want to be keeping our users and ingredients as instances of classes? if so, ingredients will be tricky because we're taking in two different arrays on instantiation 
- currently the userlogin has no validation to ensure the user input a valid username, which means the site can load without a global user and it's then really buggy, so we should maybe make it a form and slap 'required' in the html or add a function conditional
- I might be forgetting some stuff and I'll have to work with @dietza  to properly format these into issues, but take a look and let me know any questions! I think it's a good place for an initial refactor, although I could've maybe broken this branch up more.

- [x] DRY and SRP
- [x] Tested Locally

**Where should the reviewer start?**
- It's pretty much all changed, so just open the site, open the scripts, api, and dom-updates and take a peek. Start whacking buttons.

**Notes:**
- I'm happy to walk everyone through it tomorrow but hopefully it's pretty clear what I tried to do with all the naming of stuff. - The functions are in order of when the eventListener is set for it.
- I think some of the sad data needs to get dealt with in the class instantiations but I'm happy to ask Travis the best place to clean up that server data.

### CODE REVIEW

File: src/css/_recipe-display.scss [hunk 1]
```diff
@@ -189,12 +189,12 @@ main {
     opacity: 0;
   }
 
-  #icon {
+  #icon-fav, #icon-cook {
       @include transparentText($accent-secondary)
```
Comment (sschipke) [inline src/css/_recipe-display.scss:193]: Nice job using variables!

File: src/dom-updates.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    },
+
+    // HELPER FUNCTION TO INSERTHTML
+    addDisplay(area, where, what) {
+        document.querySelector(area).insertAdjacentHTML(where, what)
+    },
+
+    clearDisplay(area) {
+      document.querySelector(area).innerHTML = ''
+    },
```
Comment (sschipke) [inline src/dom-updates.js:15]: This is very smart, and I love the reusability/flexibility you have built into this very useful function!

File: src/dom-updates.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        this.addDisplay("main", "beforeend", cardHtml)
+      })
+    }, 
+
+    capitalize(words) {
+        return words.split(" ").map(word => {
+          return word.charAt(0).toUpperCase() + word.slice(1);
+        }).join(" ");
+      },
```
Comment (sschipke) [inline src/dom-updates.js:72]: You can also use CSS to make words uppercase

**Review by sschipke — APPROVED** [review]
Nice work! You've added some great functionality and really cleaned up this code!

**Review by JeffKersting — APPROVED** [review]

**Review by thaomonster — APPROVED** [review]

**Review by dietza — APPROVED** [review]
Thanks for doing so much deep work on this.
I may have questions to leave in comments but will make note of those as I am able so as to let us move forward with all of this sooner for now.

**Pull request merged**